src/app/routes/admin/form/__tests__/admin-form.routes.spec.ts

```diff
@@ -0,0 +1,268 @@
+import { ObjectId } from 'bson-ext'
+import mongoose from 'mongoose'
+import supertest, { Session } from 'supertest-session'
+
+import getUserModel from 'src/app/models/user.server.model'
+import { Status } from 'src/types'
+import { SettingsUpdateDto } from 'src/types/api'
+
+import { createAuthedSession } from 'tests/integration/helpers/express-auth'
+import { setupApp } from 'tests/integration/helpers/express-setup'
+import { buildCelebrateError } from 'tests/unit/backend/helpers/celebrate'
+import dbHandler from 'tests/unit/backend/helpers/jest-db'
+
+import { AdminFormRouter } from '../admin-form.routes'
+
+const UserModel = getUserModel(mongoose)
+
+const app = setupApp('/admin/form', AdminFormRouter, {
+  setupWithAuth: true,
+})
+
+describe('admin-form.routes', () => {
+  let request: Session
+
+  const USER_ID = new ObjectId()
+
+  beforeAll(async () => await dbHandler.connect())
+  beforeEach(async () => {
+    request = supertest(app)
+  })
+  afterEach(async () => {
+    await dbHandler.clearDatabase()
+    jest.restoreAllMocks()
+  })
+  afterAll(async () => await dbHandler.closeDatabase())
+
+  describe('PATCH /form/:formId/settings', () => {
+    it('should return 200 with latest form settings on successful update for email mode forms', async () => {
+      // Arrange
+      const { form: formToUpdate, user } = await dbHandler.insertEmailForm()
+      const session = await createAuthedSession(user.email, request)
+
+      const settingsToUpdate: SettingsUpdateDto = {
+        title: 'new title to update',
+      }
+      expect(formToUpdate.title).not.toEqual(settingsToUpdate.title)
+
+      // Act
+      const response = await session
+        .patch(`/admin/form/${formToUpdate._id}/settings`)
+        .send(settingsToUpdate)
+
+      // Assert
+      const expectedResponse = JSON.parse(
+        JSON.stringify({
```

Comment:
why stringify then immediately parse?

Reply:
Converting mongoose object to what the endpoint will actually send out, mimicking real behaviour.

For example, JSON.parse(JSON.stringify()) will remove all keys with `undefined` values (since `undefined` is not representable in the JSON spec), or convert artifacts such as `CoreMongooseArray` to `Array` types for jest to match.